app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -1,16 +1,70 @@
+#
+# Copyright 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+# require "net/http"
+# require "sys/filesystem"
+# require "uri"
+# require "json"
+# require 'fusor/password_filter'
+# require 'fusor/deployment_logger'
+
 class Fusor::Api::V21::DeploymentsController < ApplicationController
-  before_action :set_deployment, only: [:show, :update, :destroy]
+  before_action :set_deployment, only: [:show,
+                                        :update, :destroy,
+                                        :check_mount_point,
+                                        :deploy,
+                                        :redeploy,
+                                        :validate,
+                                        :log,
+                                        :openshift_disk_space,
+                                        :compatible_cpu_families]
+
+  include Fusor::Api::V21::DiscoveredHostsMixin
```

Comment:
Let's not mix in any external API calls into the deployment object and side load them.  That would be painfully slow to load data.

Instead, we need to refactor the client to only load those when necessary.  That's probably on the initial load of the application in the background without holding up the route load, or on a page where we display the data.  The deployment object can just keep it's IDs and we can set them, but we should keep the API data loads to an absolute minimum.

The only other choice would be to cache the data in our database, but that gets complex as to when to refresh that data when satellite gets modified without Fusor's knowledge.